viewing_party/party.py

```diff
@@ -0,0 +1,283 @@
+
+	user_watched_list =[]
+	friends_watched_list = []
+
+	for movie in user_data["watched"]:
+		user_watched_list.append(movie)
+
+	i = 0
+	while i < len(user_data["friends"]):
+		for movie in user_data["friends"][i]["watched"]:
+			friends_watched_list.append(movie)
+		i += 1
+	
+	# convert friend_watched_list and user_watched_list to a list of strings, then to a set of strings, and then find the difference between the 2 sets
+
+	user_watched_set = set()
+	friends_watched_set = set()
+	
+	i = 0
+	while i < len(user_watched_list):
+		user_watched_set.add(str(user_watched_list[i]))
+		i += 1
+
+	j = 0
+	while j < len(friends_watched_list):
+		friends_watched_set.add(str(friends_watched_list[j]))
+		j += 1
+
+	friends_only_watched_set = friends_watched_set - user_watched_set  
+
+	# converting difference set back to a list of dictionaries
+
+	import ast
+	friends_only_watched_list = list(friends_only_watched_set)
+	friends_only_watched_list_final = []
+
+	k = 0
+	while k < len(friends_only_watched_set):
+		convert_to_dict = ast.literal_eval(friends_only_watched_list[k])
+		friends_only_watched_list_final.append(convert_to_dict)
+		k += 1
+	
+	return friends_only_watched_list_final
+
+# *****WAVE 4******
+
+def get_available_recs(user_data):
+	
+	user_subcription_list = user_data["subscriptions"]
+	user_watched_list = user_data["watched"]
+	recommend_for_user_list = []
+
+	friends_watched_list = [] 
+	friends_watched_list_no_duplicates = []
+	friends_watched_set = set()
+
+	for friends_dict in user_data["friends"]:
+		for friends_watched_dict in friends_dict["watched"]:
+			friends_watched_list.append(friends_watched_dict)
+
+	# remove duplicates from friends_watched_list by converting 
+	# dictionaries to strings then converting outer list to a set
+
+	for watched_dict in friends_watched_list:
+		friends_watched_set.add(str(watched_dict))
+
+	# converting friends_watched_set to a list of dictionaries
+
+	import ast
+
+	for string in friends_watched_set:
+		convert_to_dict = ast.literal_eval(string)
+		friends_watched_list_no_duplicates.append(convert_to_dict)
+
+
+	for film_host_dict in friends_watched_list_no_duplicates:
+		if user_watched_list == []:
+			if film_host_dict["host"] in user_subcription_list:
+				recommend_for_user_list.append(film_host_dict)
+		else:
+			for film_dict in user_watched_list:				
+				if film_dict["title"] != film_host_dict["title"]:
+					if film_host_dict["host"] in user_subcription_list:
+						recommend_for_user_list.append(film_host_dict)
+						
+	# remove any films user has already seen from recommended movies list
+	for movie in user_watched_list:
+		for movie_host in recommend_for_user_list:
+			if movie["title"] == movie_host["title"]:
+				recommend_for_user_list.remove(movie_host)
+
+	return recommend_for_user_list
+
+# *****WAVE 5******
+
+def get_new_rec_by_genre(user_data):
```

Comment:
This works well but think about previous created functions you could use here `get_most_watched_genre(user_data) , get_friends_unique_watched(user_data), get_unique_watched(user_data)`